spec/models/reports/facility_state_spec.rb

```diff
@@ -67,4 +59,103 @@
       end
     end
   end
+
+  context "assigned patients by care states" do
+    describe "under_care, lost_to_follow_up, dead" do
+      it "computes the number of assigned patients correctly with each care state" do
+        facility = create(:facility)
+        _dead_patient = create(:patient, recorded_at: june_2021[:long_ago], assigned_facility: facility, status: "dead")
+        _under_care_patient = create(:patient, recorded_at: june_2021[:under_12_months_ago], assigned_facility: facility)
+        ltfu_patient = create(:patient, recorded_at: june_2021[:over_12_months_ago], assigned_facility: facility)
+        create(:blood_pressure, patient: ltfu_patient, recorded_at: june_2021[:over_12_months_ago])
+        _different_facility_patient = create(:patient, recorded_at: june_2021[:long_ago])
```

Comment:
Can we make a deleted patient also, ensure it's excluded?

Reply:
Can we also add a patient who was registered over 12 months ago and never came in for a BP?  They would also be LTFU.

Reply:
@harimohanraj89 I've added deleted patient/visit specs in `patient_states` and `patient_visits` tables with 7ab0ca193b173c097c2bf1e4d7dc223b2bccad46. At the facility state level the understanding is that we're not dealing with deleted patients at all, and there's nothing in the factility states matview query that deals with removing deleted patients. Do you think the test here would still be valuable?

Reply:
@rsanheim The LTFU definitions are at the `patient_state` level, and are tested there: https://github.com/simpledotorg/simple-server/blob/master/spec/models/reports/patient_state_spec.rb#L62. I was mostly testing the counting/aggregation in facility states, which is its responsibility. Wdyt?

Reply:
Ah good point, we've got good coverage of deletion in the patient state specs. Hmm, I could go either way. On one hand, adding a delete patient here makes this spec independent of the underlying implementation of facility states. On the other, relying on the patient state specs means we're assuming that facility states will always use patient states. I think that's a strong enough assumption that we can avoid adding a deleted patient here.